overlord/snapstate/storehelpers.go

```diff
@@ -767,9 +786,14 @@ func installCandidates(st *state.State, names []string, channel string, user *au
 		return nil, err
 	}
 
-	enforcedSets, err := EnforcedValidationSets(st)
-	if err != nil {
-		return nil, err
+	// if installing a specific revision, we may be trying to enforce a validation
```

Comment:
Is this code only executed on first install of a snap? And we allow this if we have a validation set saying we are not allowed to install this snap?

Reply:
It's the same situation as the other helper. If we're installing a specific revision, we don't want to check against current sets because we're installing it to make a set enforceable.